### PR TITLE
refactor(sbb-calendar): migrate component to lit

### PR DESCRIPTION
**Pull request by MarioCastigliano**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

-

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [ ] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-calendar/sbb-calendar.e2e.ts [hunk 1]
```diff
@@ -1,494 +1,483 @@
...
-    page = await newE2EPage({
-      html: `<sbb-calendar data-now="1673348400000" selected-date="${selected}"></sbb-calendar>`,
-    });
-    element = await page.find('sbb-calendar');
-    await page.waitForChanges();
+    await fixture(
+      html`<sbb-calendar data-now="1673348400000" selected-date="${selected}"></sbb-calendar>`,
+    );
+    element = document.querySelector('sbb-calendar');
+    await element.updateComplete;
```
Comment (TomMenga) [inline src/components/sbb-calendar/sbb-calendar.e2e.ts:23]:
```suggestion
    element = await fixture(
      html`<sbb-calendar data-now="1673348400000" selected-date="${selected}"></sbb-calendar>`,
    );
```

File: src/components/sbb-calendar/sbb-calendar.e2e.ts [hunk 2]
```diff
@@ -1,494 +1,483 @@
...
-    expect(await day.getAttribute('data-day')).toEqual('1 1 2023');
+    day = element.shadowRoot.querySelector('.sbb-calendar__day');
+    expect(await day.getAttribute('data-day')).to.be.equal('1 1 2023');
   });
 
   it('selects a different date', async () => {
-    const selectedSpy = await page.spyOnEvent('date-selected');
-    await page.waitForChanges();
+    const selectedSpy = new EventSpy('date-selected');
+    await element.updateComplete;
```
Comment (TomMenga) [inline src/components/sbb-calendar/sbb-calendar.e2e.ts:101]:
```suggestion
```

File: src/components/sbb-calendar/sbb-calendar.e2e.ts [hunk 3]
```diff
@@ -1,494 +1,483 @@
...
-          return document.activeElement.shadowRoot.activeElement.textContent;
-        }),
-      ).toEqual('2024');
+        (document.activeElement.shadowRoot.activeElement as HTMLElement).innerText,
+      ).to.be.equal('2024');
     });
 
     it('navigates up via keyboard', async () => {
-      await page.waitForChanges();
+      await element.updateComplete;
```
Comment (TomMenga) [inline src/components/sbb-calendar/sbb-calendar.e2e.ts:398]:
```suggestion
```

File: src/components/sbb-calendar/sbb-calendar.e2e.ts [hunk 4]
```diff
@@ -1,494 +1,483 @@
...
-          return document.activeElement.shadowRoot.activeElement.textContent;
-        }),
-      ).toEqual('2027');
+        (document.activeElement.shadowRoot.activeElement as HTMLElement).innerText,
+      ).to.be.equal('2027');
     });
 
     it('navigates to first day via keyboard', async () => {
-      await page.waitForChanges();
+      await element.updateComplete;
```
Comment (TomMenga) [inline src/components/sbb-calendar/sbb-calendar.e2e.ts:427]:
```suggestion
```

File: src/components/sbb-calendar/sbb-calendar.tsx [hunk 5]
```diff
@@ -64,50 +57,50 @@ interface CalendarKeyboardNavigationParameters {
...
+export class SbbCalendar extends LitElement {
+  public static override styles: CSSResult = Style;
+
   /** If set to true, two months are displayed */
-  @Prop() public wide = false;
+  @property({ type: Boolean }) public wide = false;
 
   /** The minimum valid date. Takes Date Object, ISOString, and Unix Timestamp (number of seconds since Jan 1, 1970). */
-  @Prop() public min: Date | string | number;
+  @property() public min: Date | string | number;
```
Comment (TomMenga) [inline src/components/sbb-calendar/sbb-calendar.tsx:72]: If the consumer uses the attribute, you'll always receive a string here when the user sets the attribute. Are you fine with it? (same thing for the other properties with the same type)

File: src/components/sbb-calendar/sbb-calendar.tsx [hunk 6]
```diff
@@ -822,209 +827,207 @@ export class SbbCalendar implements ComponentInterface {
...
-            aria-current={isToday ? 'date' : undefined}
-            data-day={dayValue}
+            })}
+            @click=${() => this._selectDate(day.value)}
+            ?disabled=${isOutOfRange || isFilteredOut}
+            aria-label=${this._dateAdapter.getAccessibilityFormatDate(day.value)}
+            aria-pressed=${selected}
+            aria-disabled=${isOutOfRange || isFilteredOut}
+            aria-current=${isToday ? 'date' : nothing}
+            data-day=${dayValue}
```
Comment (TomMenga) [inline src/components/sbb-calendar/sbb-calendar.tsx:915]: Not 100% sure that the suggestion makes sense
```suggestion
            data-day=${dayValue || nothing}
```

File: src/components/sbb-calendar/sbb-calendar.tsx [hunk 7]
```diff
@@ -822,209 +827,207 @@ export class SbbCalendar implements ComponentInterface {
...
+                      })}
+                      @click=${() => this._onMonthSelection(month.monthValue, year, shiftRight)}
+                      ?disabled=${isOutOfRange || isFilteredOut}
+                      aria-label=${`${month.longValue} ${this._chosenYear}`}
+                      aria-pressed=${selected}
+                      aria-disabled=${String(isOutOfRange || isFilteredOut)}
                       tabindex="-1"
-                      data-month={month.monthValue}
-                      onKeyDown={(evt: KeyboardEvent) => this._handleKeyboardEvent(evt)}
+                      data-month=${month.monthValue}
```
Comment (TomMenga) [inline src/components/sbb-calendar/sbb-calendar.tsx:1018]: Again, not 100% sure
```suggestion
                      data-month=${month.monthValue || nothing}
```

**Review by TomMenga — COMMENTED** [review]
Very good job, just some topic points

File: src/components/sbb-calendar/sbb-calendar.tsx [hunk 8]
```diff
@@ -1070,91 +1071,85 @@ export class SbbCalendar implements ComponentInterface {
...
+                      'sbb-calendar__crossed-out': !isOutOfRange && isFilteredOut,
+                      'sbb-calendar__selected': selected,
+                    })}
+                    @click=${() => this._onYearSelection(year, shiftRight)}
+                    ?disabled=${isOutOfRange || isFilteredOut}
+                    aria-label=${year}
+                    aria-pressed=${selected}
+                    aria-disabled=${String(isOutOfRange || isFilteredOut)}
+                    tabindex="-1"
+                    data-year=${year}
```
Comment (jeripeierSBB) [inline src/components/sbb-calendar/sbb-calendar.tsx:1142]: Is the "|| nothing" or "?" addition not needed here? (for the month you did)

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/components/sbb-calendar/sbb-calendar.tsx (hunk 5 above)
Comment (MarioCastigliano) [reply src/components/sbb-calendar/sbb-calendar.tsx:72]: I tested it with all options and they always get parsed right, so I'd say we're fine

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thanks!

**Pull request merged**